edgedbpkg/edgedb_ls/__init__.py

```diff
@@ -0,0 +1,343 @@
+from __future__ import annotations
+from typing import TYPE_CHECKING
+
+import base64
+import os
+import pathlib
+import shlex
+import textwrap
+
+from poetry.core.packages import dependency as poetry_dep
+
+from metapkg import packages
+from metapkg import targets
+from metapkg.packages import python
+
+from edgedbpkg import python as python_bundle
+from edgedbpkg import pyentrypoint
+
+if TYPE_CHECKING:
+    from cleo.io import io as cleo_io
+    from poetry.core.semver import version as poetry_version
+
+
+python.set_python_runtime_dependency(
+    poetry_dep.Dependency(
+        name="python-edgedb",
+        constraint=">=3.10.0,<3.13.0",
+        allows_prereleases=True,
+    )
+)
+
+
+class EdgeDBLanguageServer(packages.BundledPythonPackage):
```

Comment:
I think it makes sense to extend from `edgedb.EdgeDB` and remove the methods you've not actually changed as there's lots of duplicated code here.

Reply:
There is a lot of common stuff, but also slightly different. I don't think it is worth extending here because it would also add another level of inheritance which makes things less readable.